propolis/src/hw/uart/base.rs

```diff
@@ -98,10 +98,11 @@ impl Uart {
                 self.reg_intr_enable = data;
                 // Although not specified in the datasheet, some consumers expect a THRE
                 // interrupt to be raised when toggling that on in IER.
-                if old & IER_ETBEI == 0 && data & IER_ETBEI != 0 {
-                    if self.tx_fifo.is_empty() {
-                        self.thre_intr = true
-                    }
```

Comment:
I think this is more readable as-is, since it separates the register value evaluation from how the device state factors into the behavior.  The compiler will optimize this into the same result anyways.